support-frontend/assets/helpers/productCatalog.ts

```diff
@@ -65,6 +66,29 @@ export function filterBenefitByABTest(
 	return true;
 }
 
+function excludeBenefitByABTest(
+	benefit: ProductBenefit,
+	participations?: Participations,
+) {
+	if (participations && benefit.hideOnAbTest !== undefined) {
+		return benefit.hideOnAbTest.some(({ name, variants }) =>
+			participations[name]
+				? !variants.includes(participations[name] ?? '')
+				: true,
+		);
+	}
+	return true;
+}
+
+export function filterBenefitByABTest(
+	benefit: ProductBenefit,
+	participations?: Participations,
+) {
+	const includeByAbTest = includeBenefitByABTest(benefit, participations);
+	const excludeByABTest = excludeBenefitByABTest(benefit, participations);
+	return includeByAbTest && excludeByABTest;
```

Comment:
I'm interested in why we && these together, if `excludeByABTest` returns true - does that mean we shouldn't show the benefit?

Reply:
This include and exclude mechanism has a flaw you highlighted above. I have refactored with an extra field on each `benefit.specificToAbTest` called `display`, upon the test sand variant name matching you can show (`display = true`) or hide (`display = false`).